.github/workflows/build-documentation.yml

```diff
@@ -0,0 +1,22 @@
+name: build-documentation
+
+on: [push, pull_request]
```

Comment:
Consider more detailed `on`.

I'd propose the following `on` for reference: https://github.com/oneapi-src/oneTBB/blob/master/.github/workflows/ci.yml#L17

Do not forget about `release/*` branches along with `main`.

Reply:
BTW current `on` leads to double execution of the same job (see checks below).

Reply:
Updated